### DISCUSSION TITLE
How to deal with orphaned block quotes ?

### DISCUSSION THREAD

**Discussion by Limezy** [athensresearch/athens#952] [Feature Requests]
An interesting discussion was started earlier today by @yifanyin  [on the discord chanel](https://discord.com/channels/708122962422792194/800442937602342922/831124661957427230).

Block quotes are a really interesting and powerful feature. They can even be used to declutter the graph, [as explained in this article](https://costarichard.github.io/tools/roam/roam-blocks/).The problem is that block quotes in Athens are somehow "fragile", and less resilient than the page links.

### First use-case : the quoted block is not deleted, but the text is removed
**What happens ?** 
* On the quoted block, we see nothing except the number of quotes showing
* The links are not lost, and by putting again some text on the block, it will appear everywhere else

**How I think we could improve the UX ?**
* I don't think there is any point quoting a void block --> this is probably a mistake
* Moreover, on the block quote side, there is no visual indication that it's a block quote (because there is no text)
* I think there should be a small warning to the user to show him he is quoting a void block
* That warning could be just on the right of the block, like the counter for the quoted blocks
* When clicking to the warning, it could open the quoted block on the side bar, in order for the user to correct his mistake

### Second use-case : the quoted block is fully deleted
**What happens ?** 
* Every orphaned block quote will be replaced by the former quoted block address ID, which is something like `((959c7dde4))`

**How I think we could improve the UX ?**
* First proposal : 
  * We could do like Roam. Instead of replacing the former quote by its address, just replace it by the deleted quoted block content. There is some connections / linkage lost, but no text is lost
  * I also think we should give a warning to the user, like 
  * "this block was quoted X times, are you sure you want to delete it ?"
  * Same kind of warning before deleting a page that has quoted blocks inside
* Second proposal : 
  * The idea is to give the user an option to come back later on his mistake
  * For example, the former block quotes that now became addresses, like `((959c7dde4))`, would show a small warning side on the right or become clickable
  * When clicking on one of them, it would give you an option to re-create a block that has that same ID. 
  * Then you select the page on which to embed that new block
  * Once the new block is done, all former block quotes would sync again to that new block

**Comment by yifanyin** [comment]
Thanks for the impressive write-up! For the second use case, I think a combined solution like quoting the original content of the deleted block along with its ID can also work. Something like:
-  "The content within the deleted block" ((959c7dde4))

So one gets the benefit of both proposals.

**Reply by Limezy** [reply]
Yes that'd be great indeed !
